metrics/exporter.go

```diff
@@ -94,6 +191,10 @@ func setCurMetricsConfig(c *metricsConfig) {
 // Return value indicates whether the exporter is flushable or not.
 func FlushExporter() bool {
 	e := getCurMetricsExporter()
+	return flushExporterHelper(e)
```

Comment:
Could we change the lock in `getCurMetricsExporter` to read lock and not split this func? Having two is confusing. Since `UpdateExporterFromConfigMap` is not frequently used, adding a read lock won't cost too much.

Reply:
I'm not sure we can do that because a nested read lock request in UpdateExporter will still cause deadlock since writers still block out all readers.

Reply:
I agree a reader-writer lock is more appropriate here though, so I will change the mutex to be a RWMutex though and have the getters just use RLock() and setters use Lock().

It's just for FlushExporter we still need a completely unlocked helper.